### DISCUSSION TITLE
Add join and select like futures for tokio?

### DISCUSSION THREAD

**Discussion by jmjoy** [tokio-rs/tokio#3573] [Ideas] [answered]
Like these:
<https://docs.rs/futures/0.3.13/futures/index.html?search=join>
<https://docs.rs/futures/0.3.13/futures/index.html?search=select>

**Comment by Darksonn** [comment, marked as answer]
We already have the [`join!`][1], [`try_join!`][2] and [`select!`][3] macros, and you can use them to build a future by wrapping it in an async block.

If you have further arguments to providing actual structs for these, please open an issue with the feature request instead of a discussion.

[1]: https://docs.rs/tokio/1/tokio/macro.join.html
[2]: https://docs.rs/tokio/1/tokio/macro.try_join.html
[3]: https://docs.rs/tokio/1/tokio/macro.select.html

**Reply by jmjoy** [reply]
How can I join a slice of futures, such as join_all in futures?

**Reply by Darksonn** [reply]
Typically the recommendation is to use `buffer_unordered` or `FuturesUnordered` from the futures crate. (I don't recommend `join_all` due to [this](https://github.com/tokio-rs/tokio/issues/2401))
